opentracing/src/main/java/opentracing/Span.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Formatter;
+
+/**
+ * Span represents a unit of work executed on behalf of a trace. Examples of spans include a remote
+ * procedure call, or a in-process method call to a sub-component. A trace is required to have a
+ * single, top level "root" span, and zero or more children spans, which in turns can have their own
+ * children spans, thus forming a tree structure.
+ */
+public interface Span {
+
+  /**
+   * Denotes the beginning of a subordinate unit of work.
+   *
+   * @param operationName name of the operation represened by the new span from the perspective of
+   * the current service.
+   * @return a new child Span in "started" state.
+   */
+  Span startChild(String operationName);
+
+  /**
+   * Indicates that the work represented by this span has been completed or terminated, and is ready
+   * to be sent to the Reporter. <p>If any tags / logs need to be added to the span, it should be
+   * done before calling {@linkplain #finish}, otherwise they may be ignored.
+   */
+  void finish();
+
+  /**
+   * Attaches a key/value pair to the span.
+   *
+   * <p>The set of supported value types is implementation specific. It is the responsibility of the
```

Comment:
this was a copy-paste docs. I'll revise to do this with overloading as there are no union types in java.